File: python_sol/src/core/ADR.py
from src.models.operation import Operation
from src.models.register import Register

class ADR(Operation):
    @classmethod
    def execute(self, registerA : Register, registerB : Register):
        try:
            registerA.set_value(registerA.get_value() + registerB.get_value())
            return True
        except ValueError as e:
            print(e)
            return False
        

File: java_sol/src/main/java/org/opcode/model/opcodes/Adr.java
package org.opcode.model.opcodes;

import org.opcode.model.OpCodeType;
import org.opcode.model.Opcode;
import org.opcode.model.Register;

public class Adr extends Opcode<Register> {
    public Adr(Register source, Register destinationRegister){
        this.source = source;
        this.destinationRegister = destinationRegister;
        this.opcodeType = OpCodeType.ADR;
    }

    @Override
    public void execute() {
        try {
            destinationRegister.setValue(String.valueOf(source.getValue()));
        } catch (IllegalArgumentException e) {
            System.out.println("Value provided is not within limits.");
        } catch (Exception e){
            System.out.println("An error occurred while executing the Adr instruction.");
        }
    }
}
